### ISSUE TITLE
Navigation between questions (+ some functionalities)

### ISSUE THREAD

**Issue by LanderMuruaga** [exlskills/exlskills#78]
Some issues to solve on course navigation functionalities.
Need to solve interaction and design.

**Message visualization** (into question)

- [x] Show hint should be displayed and highlighted over the button to make it visible and obvious.

- [x] Show explanation should be displayed over the button to make it visible (in between the answers and the action)

- [x] Check the possibility of adding the right/wrong hint in place (into the question answer itself).

**Interaction design** (with buttons and between questions)
Submit + Next question interaction needs to be easy and obvious. Solve it. 

- [x] Submit or Check answer?

- [x] Interaction between buttons (active/inactive)

- [x] Driving users to NEXT question easily.

More on issue #76

**Comment by LanderMuruaga** [comment]
Just testing some options...
![q1](https://user-images.githubusercontent.com/41282864/46982622-47750000-d0dd-11e8-9a34-f2ec8efc5c5f.png)
![q1a](https://user-images.githubusercontent.com/41282864/46982623-47750000-d0dd-11e8-86b8-eb04e03de49c.png)
![q2](https://user-images.githubusercontent.com/41282864/46982629-5065d180-d0dd-11e8-9638-5e55d6a00f52.png)
![q3](https://user-images.githubusercontent.com/41282864/46982632-5065d180-d0dd-11e8-8989-5b8da8b80943.png)
![q4](https://user-images.githubusercontent.com/41282864/46982634-50fe6800-d0dd-11e8-8b96-2c28319c1dc1.png)
![q4a](https://user-images.githubusercontent.com/41282864/46982643-5e1b5700-d0dd-11e8-8f17-ad492e528b05.png)
![q4b](https://user-images.githubusercontent.com/41282864/46982644-5e1b5700-d0dd-11e8-8f5c-e163b7f2bafd.png)
![q5](https://user-images.githubusercontent.com/41282864/46982645-5e1b5700-d0dd-11e8-846e-1120fc27cb28.png)
![button-status](https://user-images.githubusercontent.com/41282864/46982657-6ffcfa00-d0dd-11e8-80d0-e3eba6ffa37f.png)

**Comment by LanderMuruaga** [comment]
TO CHECK:
When user checks the explanation on a question, button should be another color than green. Probably yellow following the previous pattern we created (You failed, Try again).

TO UNDERSTAND FEASIBILITY:
Another friction point could be the answer itself when user fails a question. Making checkboxes/radio-buttons **inactive** will help to avoid frustration of responding back, but this is quite secondary.

**Comment by svarlamov** [comment]
@LanderMuruaga some answers

> TO CHECK:
When user checks the explanation on a question, button should be another color than green. Probably yellow following the previous pattern we created (You failed, Try again). 

I'm not sure I see the rationale for this... The way that's it's currently implemented clearly highlights the "Next" action as the next thing to do and making it red if you get the question wrong on the card isn't really nice IMHO...

> TO UNDERSTAND FEASIBILITY:
Another friction point could be the answer itself when user fails a question. Making checkboxes/radio-buttons inactive will help to avoid frustration of responding back, but this is quite secondary.

Will do

Also, can you clarify your statements about the show/hide hint button? I don't see anything new in the screenshots you posted...

**Comment by LanderMuruaga** [comment]
@svarlamov 
> I'm not sure I see the rationale for this... The way that's it's currently implemented clearly highlights the "Next" action as the next thing to do and making it red if you get the question wrong on the card isn't really nice IMHO..

You are right. Red should not be the color. It means wrong, and users get blocked with this. 
My concern is that the green button says "You did it right" and allows you to go to next question, but in this case you didn't it right, so user needs to understand it to avoid future frustrations. 

Showing the answers inactive is one point to show that, but to make it right on our side, the button should transmit this feeling also. Probably blue is enough neutral regarding the usage we are giving to it. We can talk about it. 

![buttons](https://user-images.githubusercontent.com/41282864/47221164-fe120280-d3b3-11e8-9cb5-314f55cb2f25.png)
